keras_cv/models/object_detection/faster_rcnn.py

```diff
@@ -0,0 +1,326 @@
+        self.conv_dims = conv_dims
+        self.fc_dims = fc_dims
+        self.convs = []
+        for conv_dim in conv_dims:
+            layer = tf.keras.layers.Conv2D(
+                filters=conv_dim,
+                kernel_size=3,
+                strides=1,
+                padding="same",
+                activation="relu",
+            )
+            self.convs.append(layer)
+        self.fcs = []
+        for fc_dim in fc_dims:
+            layer = tf.keras.layers.Dense(units=fc_dim, activation="relu")
+            self.fcs.append(layer)
+        self.box_pred = tf.keras.layers.Dense(units=4)
+        self.cls_score = tf.keras.layers.Dense(units=classes + 1)
+
+    def call(self, feature_map):
+        x = feature_map
+        for conv in self.convs:
+            x = conv(x)
+        for fc in self.fcs:
+            x = fc(x)
+        rcnn_boxes = self.box_pred(x)
+        rcnn_scores = self.cls_score(x)
+        return rcnn_boxes, rcnn_scores
+
+    def get_config(self):
+        config = {
+            "classes": self.num_classes,
+            "conv_dims": self.conv_dims,
+            "fc_dims": self.fc_dims,
+        }
+        base_config = super().get_config()
+        return dict(list(base_config.items()) + list(config.items()))
+
+
+# TODO(tanzheny): add more configurations
+class FasterRCNN(tf.keras.Model):
+    def __init__(
+        self,
+        classes,
+        bounding_box_format,
+        backbone=None,
+        **kwargs,
+    ):
+        self.bounding_box_format = bounding_box_format
+        super(FasterRCNN, self).__init__(**kwargs)
+        self.anchor_generator = AnchorGenerator(
+            bounding_box_format="xyxy",
+            sizes={2: 16.0, 3: 32.0, 4: 64.0, 5: 128.0, 6: 256.0},
+            scales=[2**x for x in [0, 1 / 3, 2 / 3]],
+            aspect_ratios=[0.5, 1.0, 2.0],
+            strides={i: 2**i for i in range(2, 7)},
+            clip_boxes=True,
+        )
+        self.rpn_head = RPNHead(num_anchors_per_location=9)
+        self.roi_generator = ROIGenerator(bounding_box_format="xyxy")
+        self.box_matcher = ArgmaxBoxMatcher(
+            thresholds=[0.0, 0.5], match_values=[-2, -1, 1]
+        )
+        self.roi_sampler = _ROISampler(
+            bounding_box_format="xyxy",
+            roi_matcher=self.box_matcher,
+            background_class=classes,
+            num_sampled_rois=512,
+        )
+        self.roi_pooler = _ROIAligner(bounding_box_format="xyxy")
+        self.rcnn_head = RCNNHead(classes)
+        self.backbone = backbone or _resnet50_backbone()
+        self.feature_pyramid = FeaturePyramid()
+        self.rpn_labeler = _RpnLabelEncoder(
+            anchor_format="xyxy",
+            gt_box_format="yxyx",
+            positive_threshold=0.7,
+            negative_threshold=0.3,
+            samples_per_image=256,
+            positive_fraction=0.5,
+        )
+
+    def call(self, images, gt_boxes=None, gt_classes=None, training=None):
```

Comment:
are you planning to subclass `ObjectDetectionBaseModel` for asynchronous label encoding in tf.data pipeline?

Reply:
Not yet, still need to train a reasonable weights before inheriting.